### PR TITLE
[build-utils] Add getOutputPath and isDirectory

### PR DESCRIPTION
**Pull request by pvdz**
Ported from vc build. Need this logic in the build container too.
The logic was ported from vc-build/build.ts

### CODE REVIEW

**Review by cb1kenobi — DISMISSED** [review]
Looks good, just needs a changeset.

**Review by styfle — CHANGES_REQUESTED** [review]
This shouldn't be in `@vercel/build-utils` since that is designed for builders.

If it can go in another package, then it should since `@vercel/build-utils` is an implicit dep of all builders.

File: packages/build-utils/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # @vercel/build-utils
 
+## 7.0.1
+
+- Added `getOutputPath` to abstract the logic of determining the output folder. It will return a string which ought to be the path to the output folder.
+- Added `isDirectoryPath` which is the helper function used to determine whether a path is a directory, or not when it is a file or doesn't exist at all. It will return a boolean.
```
Comment (styfle) [inline packages/build-utils/CHANGELOG.md:6]: You shouldn't edit this file directly, its automatically generated by a bot

Comment (pvdz) [reply packages/build-utils/CHANGELOG.md:6]: ack

**Comment by pvdz** [comment]
@styfle I Was explicitly asked to put it here. Have no pref myself. Please see slack.

**Comment by pvdz** [comment]
Ok. going to add this in https://github.com/vercel/api/pull/21531 and let somebody else decide where to share this.

**Pull request closed without merging**